PyEthereum2.1 backend tests need a different environment

* Version: 0.1.0-beta.26
* OS: CI


### What was wrong?

See https://circleci.com/gh/ethereum/eth-tester/891?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

```
==================================== ERRORS ====================================
_____________ ERROR collecting tests/backends/test_pyethereum21.py _____________
tests/backends/test_pyethereum21.py:5: in <module>
    from eth_tester import (
eth_tester/__init__.py:4: in <module>
    from .main import (  # noqa: F401
eth_tester/main.py:29: in <module>
    from eth_tester.backends import (
eth_tester/backends/__init__.py:18: in <module>
    from .pyethereum.v20 import (
eth_tester/backends/pyethereum/v20/__init__.py:6: in <module>
    from .main import (  # noqa: F401
eth_tester/backends/pyethereum/v20/main.py:56: in <module>
    from ethereum.tools.tester import (
.tox/py36-pyethereum21/lib/python3.6/site-packages/ethereum/tools/__init__.py:1: in <module>
    from ethereum.tools import keys, new_statetest_utils, testutils, tester, _solidity
.tox/py36-pyethereum21/lib/python3.6/site-packages/ethereum/tools/keys.py:9: in <module>
    scrypt = __import__('scrypt')
.tox/py36-pyethereum21/lib/python3.6/site-packages/scrypt/__init__.py:1: in <module>
    from .scrypt import *
.tox/py36-pyethereum21/lib/python3.6/site-packages/scrypt/scrypt.py:15: in <module>
    _scrypt = cdll.LoadLibrary(imp.find_module('_scrypt')[1])
/usr/local/lib/python3.6/ctypes/__init__.py:426: in LoadLibrary
    return self._dlltype(name)
/usr/local/lib/python3.6/ctypes/__init__.py:348: in __init__
    self._handle = _dlopen(self._name, mode)
E   OSError: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory
```

### How can it be fixed?

Force an environment that has the libcrypto.so.1.0.0 library available. See some similar (but not exactly the same) environment issues in web3.py: https://github.com/ethereum/web3.py/commit/8327480b0f1408c59050e243d8b664b065d4c2c4

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -79,7 +79,7 @@ jobs:
   py36-pyethereum21:
     <<: *common
     docker:
-      - image: circleci/python:3.6
+      - image: circleci/python:3.6-jessie
         environment:
           TOXENV: py36-pyethereum21
   py35-pyevm:

